### PR TITLE
CONCF-989 Allow Curated Content Tool to be opened in a MediaWiki modal

### PR DESCRIPTION
**Pull request by rogatty**
https://wikia-inc.atlassian.net/browse/CONCF-989

### CODE REVIEW

**Comment by bognix** [comment]
LGTM

File: front/scripts/main/routes/CuratedContentEditorRoute.ts [hunk 1]
```diff
@@ -4,127 +4,183 @@
...
+			if (!$().cropper || !this.get('cropperLoadingInitialized')) {
+				this.suppressDefineAmd(
+					this.loadCropper()
+				);
+			}
 
-		editBlockItem(item: CuratedContentEditorItemModel, block: string): void {
-			this.trackClick('curated-content-editor', 'item-edit');
-			this.transitionTo('curatedContentEditor.blockEditItem', block, encodeURIComponent(item.label));
+			if (window.self !== window.top && (
```
Comment (tomasznapieralski) [inline front/scripts/main/routes/CuratedContentEditorRoute.ts]: can you move `&&` to new line?

Comment (rogatty) [inline front/scripts/main/routes/CuratedContentEditorRoute.ts]: Nope, see https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#conditional-logic

File: front/scripts/main/routes/CuratedContentEditorRoute.ts [hunk 2]
```diff
@@ -4,127 +4,183 @@
...
+			openSection(item: CuratedContentEditorItemModel): void {
+				this.trackClick('curated-content-editor', 'section-open');
+				this.transitionTo('curatedContentEditor.section', encodeURIComponent(item.label));
+			},
+
+			/**
+			 * Called when user clicks on custom back button or after data is saved
+			 * Does transition to the main page or sends a message through Ponto if available
+			 *
+			 * @param saved
```
Comment (tomasznapieralski) [inline front/scripts/main/routes/CuratedContentEditorRoute.ts]: wrong param

File: front/scripts/main/routes/CuratedContentEditorRoute.ts (hunk 1 above)
Comment (rogatty) [inline front/scripts/main/routes/CuratedContentEditorRoute.ts]: Nope, we keep them at the end. See https://github.com/Wikia/guidelines/blob/master/JavaScript/CodingConventions.md#multi-line-statements

>Don't start a new line with an operator.

File: front/scripts/main/routes/CuratedContentEditorRoute.ts [hunk 3]
```diff
@@ -4,127 +4,183 @@
...
+						// If something went wrong on the app side then display an error
+						// This shouldn't happen, ever
+						(err: any): void => {
+							Em.Logger.error('Ponto error:', err);
+
+							this.controllerFor('application').addAlert({
+								message: i18n.t('app.curated-content-error-other'),
+								type: 'alert'
+							});
+						}, true);
```
Comment (tomasznapieralski) [inline front/scripts/main/routes/CuratedContentEditorRoute.ts]: `true` in new line

File: front/scripts/main/routes/CuratedContentEditorRoute.ts [hunk 4]
```diff
@@ -4,127 +4,183 @@
...
+					this.transitionTo('mainPage');
+				} else {
+					Em.Logger.error(error);
+					this.controllerFor('application').addAlert({
+						message: i18n.t('app.curated-content-error-other'),
+						type: 'warning'
+					});
+					this.transitionTo('curatedContentEditor');
+				}
+				return true;
```
Comment (tomasznapieralski) [inline front/scripts/main/routes/CuratedContentEditorRoute.ts]: do you need to return `true`?

File: front/scripts/main/routes/CuratedContentEditorRoute.ts (hunk 1 above)
Comment (tomasznapieralski) [inline front/scripts/main/routes/CuratedContentEditorRoute.ts]: ok

File: front/scripts/main/routes/CuratedContentEditorRoute.ts (hunk 4 above)
Comment (bognix) [inline front/scripts/main/routes/CuratedContentEditorRoute.ts]: yes so the action bubbles higher

**Comment by hakubo** [comment]
LGTM

**Comment by vforge** [comment]
LGTM

**Pull request merged**